athina_client/services/athina_api_service.py

```
@@ -645,3 +645,80 @@ def update_dataset_cells(dataset_id: str, cells: List[Dict[str, Any]]):
+                details_message = response_json.get("details", {}).get("message", "No details")
+                raise CustomException(error_message, details_message)
+
+            return response_json["data"]["eval_run"]["id"]
+        except requests.RequestException as e:
+            raise CustomException("Request failed", str(e))
+        except Exception as e:
+            raise CustomException("Unexpected error occurred", str(e))
+
+    @staticmethod
+    @retry(stop_max_attempt_number=2, wait_fixed=1000)
+    def create_dataset_event(dataset_id: str, event_type: str, payload: Dict[str, Any], is_conditional_node: Optional[bool] = False):
```

Ruff (0.8.2)

Undefined name `Optional`

(F821)